test/contrib/test_socks.py

```diff
@@ -294,16 +311,17 @@ def request_handler(listener):
             assert response.data == b""
             assert response.headers["Server"] == "SocksTestServer"
 
-    def test_correct_header_line(self):
-        def request_handler(listener):
+    def test_correct_header_line(self) -> None:
+        def request_handler(listener: socket.socket) -> None:
             sock = listener.accept()[0]
 
             handler = handle_socks5_negotiation(sock, negotiate=False)
             addr, port = next(handler)
 
             assert addr == b"example.com"
             assert port == 80
-            handler.send(True)
+            with pytest.raises(StopIteration):
```

Comment:
Why is this added?

Reply:
The original implementation of `handle_socks*_negotiation` would yield multiple types of varying types. Sometimes a (addr, port) tuple and other times a bool. The bool looks like it was added to avoid handling a `StopIteration` exception.The yielded value is not actually used. We could define the generator to yield `Union[Tuple, bool]` however, this complicates the call sites as now the type checker will complain that `bool` is not iterable. I found it simpler to only yield a single type `(addr, port)` and in other code branches to stop the generator, hence `StopIteration`.

Is there a preference for a different approach?

Reply:
Your approach is much better, thanks for making this change :+1: